src/TitleReceive/TitleReceiveContainer.js

```diff
@@ -55,7 +60,15 @@ function TitleReceiveContainer({ history, location, match, mutator, resources })
       if (poLineId) {
         mutator.poLine.GET({
           path: `${PO_LINES_API}/${poLineId}`,
-        }).then(setPoLine);
+        }).then(poLineResponse => {
+          setPoLine(poLineResponse);
+
+          const lineLocationIds = poLineResponse?.locations?.map(({ locationId }) => locationId);
+
+          setPoLineLocationIds(lineLocationIds);
+
+          return batchFetch(mutator.pieceLocations, lineLocationIds);
```

Comment:
IMHO, this should be similar to Add Piece - dictionary of locations could contain all po line locations and all pieces locations, so common component will not be making similar API requests for same locations for pieces.

Reply:
In short: you're doing request twice with location ids that might overlap. The idea was to do API request for locations once, with ids of po line and all pieces locations

Reply:
Added new `useEffect` for fetching locations